client/lcd/certificates.go

```diff
@@ -0,0 +1,163 @@
+package lcd
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"os"
+	"strings"
+	"time"
+)
+
+// default: 10 years
+const defaultValidFor = 365 * 24 * time.Hour
```

Comment:
Can anyone think a reason why 10 year validity is dangerous for a ephemeral cert?

In most cases where you would have a unusually long lived instance of the LCD best practice would be to supply your own Cert.

Reply:
@zmanian  I've set it to 1 year already. Do you reckon that is reasonable?

Reply:
Maybe 30 days? @jessysaurusrex What do you think?

Reply:
Certainly not a year either I think. 30 days sounds reasonable. Some other system event might even kill the LCD before that anyway.